cmd/cloudshell_open/main.go

```diff
@@ -343,6 +358,15 @@ func run(opts runOpts) error {
 		return err
 	}
 
+	if reflect.DeepEqual(existingService, service{}) {
+		for _, command := range appFile.Hooks.PostCreate.Commands {
+			err = runScript(appDir, command, hookEnvs)
+			if err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
can we refactor these hooks out, to something like


```
if err := executeHooks("postcreate", appFile.Hooks.PostCreate.Commands); err != nil { return err }
```

you can construct the detailed error message in `executeHooks`.

Reply:
this would also

- simplify/shorten this method
- prevent duplicating this logic (e.g. for loops around runScript)